wsmaster/che-core-api-devfile/src/main/resources/schema/devfile.json

```diff
@@ -92,6 +92,11 @@
                 "examples": [
                   "git@github.com:spring-projects/spring-petclinic.git"
                 ]
+              },
+              "refspec": {
+                "type": "string",
+                "description": "The name of the refspec to check out after the clone. This can be a branch name, a tag or commit id.",
```

Comment:
Type can be `zip` as it is shown in examples for type field. Can you describe behavior in such case? Will it be just ignored? Will an exception be thrown that it is not supported for specified source type?

Reply:
It actually depends on the Theia side, IMHO. Serverside doesn't do anything with this attribute but translate it back and forth between workspace config and devfile. As far as I looked, Theia only supports git ATM. @sunix, @benoitf could you please add more details to this?

Reply:
Does it makes sense to add the same as you add to java docs of sources, like
> This can be a branch, tag, commit id or anything the particular source type understands.

Reply:
:+1:

Reply:
I've updated the description. That said, I still cannot fully answer your question. The behavior is source type specific and is implemented in Theia. So on the server side we cannot tell whether this is going to fail the workspace or will be ignored.